File: electron/src/electron-utils/utilsMigrate.ts
import { UtilsFile } from './utilsFile';
export class UtilsMigrate {
  private _uFile: UtilsFile = new UtilsFile();

  public addSQLiteSuffix(folderPath: string): Promise<void> {
    return new Promise(async (resolve, reject) => {
      const toPath: string = this._uFile.getDatabasesPath();
      folderPath =
        folderPath !== 'default'
          ? this._uFile.getCustomerPath(folderPath)
          : toPath;
      // check if path exists
      const isPath: boolean = this._uFile.isPathExists(folderPath);
      if (!isPath) {
        reject(new Error(`path ${folderPath} does not exists`));
      }
      // get the files
      const files: string[] = await this._uFile.getFileList(folderPath);
      files.forEach(async file => {
        let newFile: string = file;
        if (!file.includes('SQLite.db')) {
          const arFile: string[] = file.split('.db');
          if (arFile.length != 2) {
            reject(new Error('Not a .db file'));
          }
          newFile = arFile[0].concat('SQLite.db');
          try {
            await this._uFile.copyFile(folderPath, file, toPath, newFile);
          } catch (err) {
            reject(new Error(`addSQLiteSuffix: ${err.message} `));
          }
        }
      });
      resolve();
    });
  }
  public deleteOldDatabases(folderPath: string): Promise<void> {
    return new Promise(async (resolve, reject) => {
      const toPath: string = this._uFile.getDatabasesPath();
      folderPath =
        folderPath !== 'default'
          ? this._uFile.getCustomerPath(folderPath)
          : toPath;
      // check if path exists
      const isPath: boolean = this._uFile.isPathExists(folderPath);
      if (!isPath) {
        reject(new Error(`path ${folderPath} does not exists`));
      }
      console.log(`$$folderPath ${folderPath}`);
      // get the files
      const files: string[] = await this._uFile.getFileList(folderPath);
      files.forEach(async file => {
        if (!file.includes('SQLite.db')) {
          try {
            console.log(`$$file ${file}`);
            const fPath: string = this._uFile.getCustomerFilePath(
              folderPath,
              file,
            );
            await this._uFile.deleteFilePath(fPath);
          } catch (err) {
            reject(new Error(`deleteFilePath:  ${folderPath} failed`));
          }
        }
      });
      resolve();
    });
  }
}

File: android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsMigrate.java
package com.getcapacitor.community.database.sqlite.SQLite;

import android.content.Context;
import com.getcapacitor.community.database.sqlite.SQLite.UtilsFile;
import java.io.File;

public class UtilsMigrate {

    private static final String TAG = UtilsMigrate.class.getName();
    private UtilsFile uFile = new UtilsFile();

    public void addSQLiteSuffix(Context context, String folderPath) throws Exception {
        String pathDB = new File(context.getFilesDir().getParentFile(), "databases").getAbsolutePath();
        String pathFiles = this.getFolder(context, folderPath);
        // check if the path exists
        File dir = new File(pathFiles);
        if (!dir.exists()) {
            throw new Exception("Folder " + dir + " does not exist");
        }
        String[] listFiles = dir.list();
        for (String file : listFiles) {
            if (uFile.getFileExtension((file)).equals("db")) {
                if (!file.contains("SQLite.db")) {
                    String fromFile = file;
                    String toFile = file.replace(".db", "SQLite.db");
                    boolean ret = uFile.copyFromNames(context, pathFiles, fromFile, pathDB, toFile);
                    if (!ret) {
                        String msg = "Failed in copy " + fromFile + " to " + file;
                        throw new Exception(msg);
                    }
                }
            }
        }
        return;
    }

    public String getFolder(Context context, String folderPath) throws Exception {
        String pathFiles = context.getFilesDir().getAbsolutePath();
        String pathDB = new File(context.getFilesDir().getParentFile(), "databases").getAbsolutePath();
        if (!folderPath.equals("default")) {
            String[] arr = folderPath.split("/", 2);
            if (arr.length == 2) {
                if (arr[0].equals("files")) {
                    pathFiles = pathFiles.concat("/").concat(arr[1]);
                } else if (arr[0].equals("databases")) {
                    pathFiles = pathDB.concat("/").concat(arr[1]);
                } else {
                    throw new Exception("Folder " + folderPath + " not allowed");
                }
            }
        } else {
            pathFiles = pathDB;
        }
        return pathFiles;
    }

    public void deleteOldDatabases(Context context, String folderPath) throws Exception {
        String pathFiles = this.getFolder(context, folderPath);
        // check if the path exists
        File dir = new File(pathFiles);
        if (!dir.exists()) {
            throw new Exception("Folder " + dir + " does not exist");
        }
        String[] listFiles = dir.list();
        for (String file : listFiles) {
            if (uFile.getFileExtension((file)).equals("db")) {
                if (!file.contains("SQLite.db")) {
                    boolean ret = uFile.deleteFile(pathFiles, file);
                    if (!ret) {
                        String msg = "Failed in delete " + file;
                        throw new Exception(msg);
                    }
                }
            }
        }
        return;
    }
}
